server/models/meshmodel/registration/utils.go

```diff
@@ -0,0 +1,57 @@
+package registration
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha2"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1beta1"
+	"github.com/layer5io/meshkit/models/meshmodel/entity"
+	"gopkg.in/yaml.v2"
+)
+
+func unmarshal(byt []byte, out interface{}) error {
+	err := json.Unmarshal(byt, out)
+	if(err != nil){
+		err = yaml.Unmarshal(byt, out)
+		if(err != nil){
+			return fmt.Errorf("Not a valid YAML or JSON")
+		}
+	}
+	return nil
+}
+
+// TODO: refactor this and use CUE
+func getEntity(byt []byte) (et entity.Entity, _ error) {
+    var versionMeta v1beta1.VersionMeta
+	err  := unmarshal(byt, &versionMeta)
+	if err != nil || versionMeta.SchemaVersion == "" {
+		return nil, ErrGetEntity(fmt.Errorf("Does not contain versionmeta"))
```

Comment:
versionmeta?

Reply:
This `versionMeta` is just an internal struct that will not be propagated to the user. This is just a helper struct that holds `schemaVersion` and `version`.